### ISSUE TITLE
GLIDE: Towards Photorealistic Image Generation and Editing with Text-guided Diffusion Models

### ISSUE THREAD

**Issue by chaerinkong** [reyllama/paper-reviews#161]
## TL; DR

- Scales up diffusion models for text-conditional image generation task.

## Motivation

- Diffusion models have shown to outperform previous SOTA methods on various benchmarks.
- Recently, classifier-guidance and classifier-free guidance methods have significantly improved class-conditional generation performance of diffusion models.

## Prior Works

- DDPM
- Improved DDPM
- DDIM
- ADM (Diffusion models beat GANs)
- Classifier-free Diffusion Models
- Palette

### Classifier-free Guidance

- Replace text condition with zero vector with fixed probability (0.2 during fine tuning step)
- Extrapolate between conditional and unconditional model prediction.

### CLIP-guidance

- Employ clip score (inner product between text and image embedding) as classifier logit (apply log gradient as usual)
- Since the classifier has to be trained on *noised* images (otherwise performance drop due to OOD) train noised-CLIP separately.

### Model Composition

- Text Encoder (transformer, 1.2B)
- Text-to-Image Model (diffusion, 2.3B, 64x64)
- Image upsampler (diffusion with low-res image condition, 1.5B, 256x256)

### Training

- Pretrain the models (2.5M iters with batch 2048 for base T2I / 1.6M iters with batch 512 for the upsampler)
- Fine-tune for classifier free guidance *after pretrianing*, with text embeddings replaced by empty sequences 20% of the time.
- Fine tune for inpainting, where additional mask that indicates regions to be inpainted are channel-wise concatenated and provided.
- ViT CLIP model is trained on noised images of resolution 64x64.

## Experiments

![image](https://user-images.githubusercontent.com/46308407/153735738-9777cb2f-a213-4a15-af64-c3077826632c.png)
![image](https://user-images.githubusercontent.com/46308407/153735740-9d5f2dd2-35a4-4bac-b95d-ee2bb92eef55.png)
![image](https://user-images.githubusercontent.com/46308407/153735743-f8da82a2-5d2e-4eb5-96b7-a8d6c137a76a.png)
![image](https://user-images.githubusercontent.com/46308407/153735748-ec1c44a6-8754-4875-b4cb-5680468c3076.png)
![image](https://user-images.githubusercontent.com/46308407/153735753-6795e3b5-f209-4539-afb7-623c95168c49.png)
![image](https://user-images.githubusercontent.com/46308407/153735756-2af127bc-acf7-46fc-80df-b7a49e1fba0e.png)
![image](https://user-images.githubusercontent.com/46308407/153735761-506a2dd5-f2b7-46b5-b294-e415647f13b7.png)

- Impressive quality for zero-shot text-to-image synthesis.
- Classifier-free guidance is preferred over CLIP-guidance over multiple standards. The only metric CLIP-guidance displays superiority is the CLIP score, which the authors hypothesize is due to adversarial training effect on CLIP.

## Comments

- Impressive quality, and the part that the authors explicitly incorporated image inpainting for iterative/interactive image generation is especially interesting.
- It might be true that diffusion models are much easier / more straight-forward to train compared to other generative models like GANs. The whole model trained stably with 16-bit precision, per the authors.
- I have a feeling that this year's CVPR (CVPR'22) is going to be dominated by diffusion models, at least in generative models domain.
